### ISSUE TITLE
Reddit search issue

### ISSUE THREAD

**Issue by marymlucas** [JustAnotherArchivist/snscrape#317]
I'm using the following command in the shell/terminal:  

snscrape --jsonl --progress reddit-search "long covid" > longc.json 

Hoping to get posts that have that exact phrase. However some of the results i get back have the two words separately but not the exact phrase.  What could be happening here?

**Comment by TheTechRobo** [comment]
It is because you are using the double-quotes. Try `snscrape --jsonl --progress reddit-search "\"long covid\""`.

**Comment by JustAnotherArchivist** [comment]
That's correct. Minor addition: Escaping the double quotes *and the space* would work, i.e. `\"long\ covid\"` (otherwise you'd get two arguments `"long` and `covid"`). Alternatively, you could also wrap it all in single quotes, i.e. `'"long covid"'`.

All of this happens in your shell. snscrape needs to see `"long covid"` as the final argument, and those are different ways of achieving that despite the shell normally interpreting quotes for its own purposes. I'm assuming a standard sane shell here (sh, bash, zsh, etc.); the Windows terminal has different quoting rules.

**Comment by marymlucas** [comment]
Thank you for the replies. That makes sense.  Out of curiosity, is this purely a double quote issue? Would just using single quotes work? 

snscrape --jsonl --progress reddit-search 'long covid' > longc.json

I'm assuming this is something I should keep in mind regardless of whether I'm scraping Reddit or Twitter. 

I'm using a standard Linux shell, the default one that comes with Pop OS.

**Comment by JustAnotherArchivist** [comment]
It works but has a different meaning. `'long covid'` and `"long covid"` in the shell both lead to a single argument `long covid` being supplied to snscrape. Which then leads to results containing both words but not necessarily the phrase.

When you use `'"long covid"'` (or one of the other variants above) in the shell, snscrape gets `"long covid"` as the argument and passes that on to the Pushshift API used for Reddit data retrieval. That server then interprets the quotes as a phrase to search for as opposed to two separate words.

Basically, the quotes have an effect in the shell, and if you want them to be passed in the argument to the program you're executing (like snscrape in this case), you have to escape them somehow. Single and double quotes also have a different meaning in shells, by the way. I'd recommend a general tutorial about the shell you're using (run `echo $SHELL` inside one to see) to understand it better.

**Comment by marymlucas** [comment]
Thank you for taking the time to explain all that! Very helpful!
